airbyte-integrations/connectors/source-salesforce/unit_tests/api_test.py

```diff
@@ -446,50 +440,47 @@ def test_csv_reader_dialect_unix():
         assert result == data
 
 
-@patch("source_salesforce.source.BulkSalesforceStream._non_retryable_send_http_request")
-def test_given_retryable_error_when_download_data_then_retry(send_http_request_patch):
-    send_http_request_patch.return_value.iter_content.side_effect = [HTTPError(), _A_CHUNKED_RESPONSE]
+@pytest.fixture(name="mocked_response")
+def _create_mocked_response():
+    http_response = Mock()
+    http_response.headers = {}
+    return http_response
+
+
+@patch("source_salesforce.streams.HttpClient")
+def test_given_retryable_error_when_download_data_then_retry(mocked_http_client, mocked_response):
+    mocked_http_client.return_value.send_request.return_value = (Mock(), mocked_response)
+    mocked_response.iter_content.side_effect = [ChunkedEncodingError(), _A_CHUNKED_RESPONSE]
+
     BulkSalesforceStream(stream_name=_A_STREAM_NAME, sf_api=Mock(), pk=_A_PK).download_data(url="any url")
-    assert send_http_request_patch.call_count == 2
+
+    assert mocked_response.iter_content.call_count == 2
 
 
-@patch("source_salesforce.source.BulkSalesforceStream._non_retryable_send_http_request")
-def test_given_first_download_fail_when_download_data_then_retry_job_only_once(send_http_request_patch):
+@patch("source_salesforce.streams.HttpClient")
+def test_given_first_download_fail_when_download_data_then_retry_job_only_once(mocked_http_client, mocked_response):
     sf_api = Mock()
     sf_api.generate_schema.return_value = JobInfoResponseBuilder().with_state("JobComplete").get_response()
     sf_api.instance_url = "http://test_given_first_download_fail_when_download_data_then_retry_job.com"
     job_creation_return_values = [_A_JSON_RESPONSE, _A_SUCCESSFUL_JOB_CREATION_RESPONSE]
-    send_http_request_patch.return_value.json.side_effect = job_creation_return_values * 2
-    send_http_request_patch.return_value.iter_content.side_effect = HTTPError()
+
+    mocked_http_client.return_value.send_request.return_value = (Mock(), mocked_response)
+    mocked_response.json.side_effect = job_creation_return_values * 2
+    mocked_response.iter_content.side_effect = ChunkedEncodingError()
 
     with pytest.raises(Exception):
         list(BulkSalesforceStream(stream_name=_A_STREAM_NAME, sf_api=sf_api, pk=_A_PK).read_records(SyncMode.full_refresh))
 
-    assert send_http_request_patch.call_count == (len(job_creation_return_values) + _NUMBER_OF_DOWNLOAD_TRIES) * 2
-
-
-@patch("source_salesforce.source.BulkSalesforceStream._non_retryable_send_http_request")
-def test_given_http_errors_when_create_stream_job_then_retry(send_http_request_patch):
```

Comment:
This test is covered by `test_given_job_creation_have_transient_error_when_read_then_sync_properly`